packages/core/src/order/order.ts

```diff
@@ -13,6 +13,10 @@ type OrderType = 'asc' | 'desc';
 export class Order<EntityT extends Entity>
   implements EntityIdentifiable<EntityT> {
   readonly _entityConstructor: Constructable<EntityT>;
+  /**
+   * Dummy property to make the type checks work.
+   */
+  protected makeDifferent: EntityT;
```

Comment:
Two points here:
1. `makeDifferent` is a name that is difficult to understand.
2. This is what EntityIdentifiable should do. If this is not working `EntityIdentifiable` should be fixed centrally. I assume this would save us from implementing that everywhere specifically?

Reply:
I try it with EntityIdentifiable, Contructable does not have a property that is right, but methods having it as a return type. I adjusted the  class to have a static _Entity property and added it to the constructable but this did not work.